sql-plugin/src/main/scala/org/apache/spark/sql/rapids/arithmetic.scala

```diff
@@ -282,6 +282,54 @@ object GpuDivModLike {
       case t => throw new IllegalArgumentException(s"Unexpected type: $t")
     }
   }
+
+  /**
+   * This is for the case as below.
+   *
+   *   left : [1,  2,  Long.MinValue,  3, Long.MinValue]
+   *   right: [2, -1,             -1, -1,             6]
+   *
+   * The 3rd row (Long.MinValue, -1) will cause an overflow of the integral division.
+   */
+  def isDivOverflow(left: GpuColumnVector, right: GpuColumnVector): Boolean = {
+    left.dataType() match {
+      case LongType =>
+        withResource(Scalar.fromLong(Long.MinValue)) { minLong =>
+          withResource(left.getBase.equalTo(minLong)) { eqToMinLong =>
+            withResource(Scalar.fromInt(-1)) { minusOne =>
+              withResource(right.getBase.equalTo(minusOne)) { eqToMinusOne =>
+                withResource(eqToMinLong.and(eqToMinusOne)) { overFlowVector =>
+                  withResource(overFlowVector.any()) { isOverFlow =>
```

Comment:
@jlowe what is the behavior of any around nulls? I thought that there were some issues?

Reply:
`any()` like `all()` will skip null values, but annoyingly it will return an invalid scalar if the column has no other valid values.
```
scala> val c = ai.rapids.cudf.ColumnVector.fromBoxedBooleans()
c: ai.rapids.cudf.ColumnVector = ColumnVector{rows=0, type=BOOL8, nullCount=Optional[0], offHeap=(ID: 53 7fa0a9fafed0)}

scala> c.any().isValid
res8: Boolean = false

scala> val c = ai.rapids.cudf.ColumnVector.fromBoxedBooleans(null, null, null)
c: ai.rapids.cudf.ColumnVector = ColumnVector{rows=3, type=BOOL8, nullCount=Optional[3], offHeap=(ID: 60 7fa0a8494670)}

scala> c.any().isValid
res9: Boolean = false
```

I filed https://github.com/rapidsai/cudf/issues/8757 a while back to track this unexpected behavior from cudf.